src/main.rs

```diff
@@ -33,29 +35,40 @@ use crate::{
     util::json,
 };
 
-fn start_logger(default_level: LevelFilter) {
+fn start_logger(default_level: LevelFilter, with_syslog: bool) {
     let filter = match EnvFilter::try_from_default_env() {
-        Ok(filter) => filter
-            .add_directive("hyper=off".parse().unwrap())
-            .add_directive("ethers=off".parse().unwrap()),
-        _ => EnvFilter::default()
-            .add_directive(default_level.into())
-            .add_directive("hyper=off".parse().unwrap())
-            .add_directive("ethers=off".parse().unwrap()),
+        Ok(filter) => filter,
+        _ => EnvFilter::default().add_directive(default_level.into()),
     };
-
-    tracing_subscriber::fmt()
-        .with_env_filter(filter)
-        .with_target(false)
-        .init();
+    let filter = filter
+        .add_directive("hyper=off".parse().unwrap())
+        .add_directive("ethers=off".parse().unwrap());
```

Comment:
```suggestion
    filter.add_directive("hyper=off".parse().unwrap())
          .add_directive("ethers=off".parse().unwrap());
```
I don't think we need to shadow the original variable here

Reply:
well, that doesn't compile:
```
   Compiling state-reconstruct v0.1.0 (/home/vb/zksync-state-reconstruct)
error[E0382]: use of moved value: `filter`
  --> src/main.rs:49:47
   |
39 |     let filter = match EnvFilter::try_from_default_env() {
   |         ------ move occurs because `filter` has type `EnvFilter`, which does not implement the `Copy` trait
...
43 |     filter
   |     ------ value moved here
...
49 |     let subscriber = Registry::default().with(filter).with(default_layer);
   |                                               ^^^^^^ value used here after move

For more information about this error, try `rustc --explain E0382`.
error: could not compile `state-reconstruct` (bin "state-reconstruct" test) due to previous error
```

Reply:
Hmm, is there any reason we can't keep it as it was before then? I.e.
```rust
    let filter = match EnvFilter::try_from_default_env() {
        Ok(filter) => filter
            .add_directive("hyper=off".parse().unwrap())
            .add_directive("ethers=off".parse().unwrap()),
        _ => EnvFilter::default()
            .add_directive(default_level.into())
            .add_directive("hyper=off".parse().unwrap())
            .add_directive("ethers=off".parse().unwrap()),
    };

    let default_layer = tracing_subscriber::fmt::layer().with_target(false);
    let subscriber = Registry::default().with(filter).with(default_layer);

```

Reply:
I guess so, in my opinion that's not such a big deal, but I'll leave this for you to decide